docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1040 @@
+
+A `Category` object holds the following attributes:
+* A `String` object that holds the category name in **lower-case** format.
+* A `BookList` object that stores a list of `Book` objects tagged with the specified category name.
+* A `QuoteList` object that stores a list of `Quote` objects tagged with the specified category name.
+* An `int` value that indicates the total number of books/quotes tagged under the specified category.
+
+#### 4.4.1 Add Categories
+The proposed add categories feature allows a user to add multiple categories to an existing book, quote, or both. 
+
+The sequence diagram below demonstrates the command execution process when adding a category to an existing book.
+
+![Sequence Diagram for Add Categories](images/SeqDiagram_AddCategories.png)
+
+* For each category that the user has specified, the process will be executed in a loop until all categories have been processed.
+* Additional checks include verifying the existence of the specified book and if the specified category already exists in the book. If either one of these checks fail, an error message will be prompted.
+* On success, 
+  * The category name will be added into the *categories* attribute of the `Book` object. 
+  * A new `Category` object will be created and stored into the category list if it does not exist. 
+  * The book will be added into the category's *bookList* attribute for record keeping.
+
+##### Design Consideration
+* Allowing users to specify multiple categories at once.
+  * Pros: Increases efficiency for users
+* Giving users an option to specify a book, quote, or both to be tagged with a category.
+  * Pros: Increases efficiency for users.
+  * Cons: Difficult to implement.
+* Giving users an option to specify *multiple* books, quotes, or both to be tagged with a category:
+  * Pros: Further increases efficiency for users.
+  * Cons: Increased complexity in implementation. 
+
+### 4.5 Feature: Rating system for books
+Given below is the class diagram for classes related to the Rating System in Quotesify:
+
+![Class Diagram for Rating system](images/ClassDiagram_Rating.png)
+
+* A `Book` must exists before a `Rating` can be created.
+* The `XRatingCommand` class represents `AddRatingCommand`, `DeleteRatingCommand`, `ListRatingCommand`
+, `EditRatingCommand`, `FindRatingCommand` classes which all extends their respective parent `XCommand` classes.
+* All the `XCommand` classes extends the abstract `Command` class.
+
+#### 4.5.1 Add rating
+The *add rating* feature will rely on an existing book object, and a rating object will then be created
+in the process.
+* The book object will store an attribute named *rating*, which will be set by this feature.
+* The rating object will be made up of the book object and a rating score, which is stored in a list of ratings.
+
+Given below is the sequence diagram for adding rating to a book:
+
+![Sequence Diagram for Add Ratings](images/SeqDiagram_AddRating.png)
```

Comment:
![image](https://user-images.githubusercontent.com/57656916/97528343-71f63500-19e8-11eb-9f92-7092c566c322.png)
should the 'X' be lower in position from operation box such that it would not confuse the reader?

Reply:
Perhaps, could you provide 'X' mark after each operation ends such that it tells clearly where the operation stops?